cypress/e2e/downloads.cy.ts

```diff
@@ -0,0 +1,369 @@
+        cy.get('[data-testid="download-overlay-button"]').click({force: true});
+        cy.get('[data-testid="download-overlay"]').should('exist');
+      });
+    });
+  });
+
+  describe('download overlay', () => {
+    describe('close button', () => {
+      beforeEach(() => {
+        cy.intercept('**/playManifest/**', {fixture: 'video.mp4'});
+      });
+
+      it('should hide overlay on click', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+          cy.get('[data-testid="download-overlay-button"]').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('exist');
+          cy.get('[data-testid="download-overlay-close-button"]').should('exist').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+        });
+      });
+    });
+
+    describe('download button', () => {
+      beforeEach(() => {
+        cy.intercept('**/playManifest/**', {fixture: 'video.mp4'}).as('playManifest');
+      });
+
+      it('should hide overlay on click', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+          cy.get('[data-testid="download-overlay-button"]').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('exist');
+          cy.get('[data-testid="download-overlay-download-button"]').should('exist').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+        });
+      });
+
+      it('should start download on click', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+          cy.get('[data-testid="download-overlay-button"]').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('exist');
+          cy.get('[data-testid="download-overlay-download-button"]').should('exist').click({force: true});
+
+          setTimeout(() => {
+            const downloadsFolder = Cypress.config('downloadsFolder');
+            const downloadPath = `${downloadsFolder}/download.mp4`;
+            cy.readFile(downloadPath);
+          }, 2000);
+        });
+      });
+
+      it('should call pre-download hook', () => {
+        let count = 0;
+
+        const preDownloadHook = () => {
+          ++count;
+        };
+        const pluginConfig = {
+          download: {preDownloadHook},
+          uiManagers: {}
+        };
+
+        return loadPlayerAndSetMedia({plugins: pluginConfig}).then(() => {
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+          cy.get('[data-testid="download-overlay-button"]').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('exist');
+          cy.get('[data-testid="download-overlay-download-button"]').should('exist').click({force: true});
```

Comment:
I believe you could use `then` promise instead of setTimeout
`cy.get('[data-testid="download-overlay-download-button"]').should('exist').click({force: true}).then(() => {
expect(count).to.equal(1);
})`

Reply:
I've checked, cypress didn't actually execute the then block in this case